### ISSUE TITLE
Catalina Python3 Support

### ISSUE THREAD

**Issue by Integralist** [vim/vim#7112]
**Describe the bug**

Manually compiling Vim with Python3 support results in Vim being compiled and installed, but without Python3 support.

**To Reproduce**

I'm on macOS Catalina with Homebrew installed.

- `brew install pyenv`
- `pyenv install 3.9.0`
- `pyenv global 3.9.0`
- `mkdir Code && cd Code`
- `git clone https://github.com/vim/vim.git`
- `cd vim`
- `make clean distclean`
- `./configure --with-features=huge --enable-multibyte --enable-rubyinterp=yes --enable-python3interp=yes --enable-perlinterp=yes --enable-luainterp=yes --enable-gui=gtk2 --enable-cscope --prefix=/usr/local`
- `make && make install`

**Expected behavior**
Vim to be compiled with Python3 support.

**Environment (please complete the following information):**
 - OS: macOS Catalina 10.15.4
 - Terminal: macOS terminal

**Additional context**
Here is a partial view of the compilation output that shows where it's checking for Python3...

```
checking --enable-pythoninterp argument... no
checking --enable-python3interp argument... yes
checking --with-python3-command argument... no
checking for python3... /Users/integralist/.pyenv/shims/python3
checking Python version... 3.9
checking Python is 3.0 or better... yep
checking Python's abiflags... 
checking Python's install prefix... /Users/integralist/.pyenv/versions/3.9.0
checking Python's execution prefix... /Users/integralist/.pyenv/versions/3.9.0
checking Python's configuration directory... /Users/integralist/.pyenv/versions/3.9.0/lib/python3.9/config-3.9-darwin
checking Python3's dll name... libpython3.9.a
checking if -pthread should be used... no
checking if compile and link flags for Python 3 are sane... yes
checking if -fPIE can be added for Python3... yes
```

But later on we see a compilation error...

```
gcc -c -I. -I/Users/integralist/.pyenv/versions/3.9.0/include/python3.9 -fPIE  -Iproto -DHAVE_CONFIG_H   -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/if_python3.o if_python3.c
if_python3.c:1009:2: warning: 'PyEval_InitThreads' is deprecated [-Wdeprecated-declarations]
        PyEval_InitThreads();
        ^
/Users/integralist/.pyenv/versions/3.9.0/include/python3.9/ceval.h:130:1: note: 'PyEval_InitThreads' has been explicitly marked deprecated here
Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
^
/Users/integralist/.pyenv/versions/3.9.0/include/python3.9/pyport.h:508:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
```

**Comment by Integralist** [comment]
Hmm, I tried adding `--with-python3-command=/Users/integralist/.pyenv/shims/python` and I still get the above error about `PyEval_InitThreads` _but_ I do now see `+python3` when checking the Vim version (also the Ex command yields expected output)

```
:py3 import sys; print(sys.version)
```

**Comment by k-takata** [comment]
> But later on we see a compilation error...

It's not an error but a warning, and it should have been fixed in 8.2.1834.
And it seems that you can successfully compile vim by using `--with-python3-command` configure option. So, closing.
